nexus/tests/integration_tests/images.rs

```diff
@@ -0,0 +1,462 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Tests images support in the API
+
+use http::method::Method;
+use http::StatusCode;
+use nexus_test_utils::http_testing::AuthnMode;
+use nexus_test_utils::http_testing::NexusRequest;
+use nexus_test_utils::http_testing::RequestBuilder;
+use nexus_test_utils::resource_helpers::create_organization;
+use nexus_test_utils::resource_helpers::create_project;
+use nexus_test_utils::resource_helpers::DiskTest;
+use nexus_test_utils::ControlPlaneTestContext;
+use nexus_test_utils_macros::nexus_test;
+
+use omicron_common::api::external::{ByteCount, IdentityMetadataCreateParams};
+use omicron_nexus::external_api::params;
+use omicron_nexus::external_api::views::GlobalImage;
+
+use httptest::{matchers::*, responders::*, Expectation, ServerBuilder};
+
+#[nexus_test]
+async fn test_global_image_create(cptestctx: &ControlPlaneTestContext) {
+    let client = &cptestctx.external_client;
+    DiskTest::new(&cptestctx).await;
+
+    let server = ServerBuilder::new().run().unwrap();
```

Comment:
What exactly are we testing here, with this use of `httptest`? As far as I can tell, these tests:

- Create a mock server via `ServerBuilder::new()`
- Set up some expectations via `.expect()`
- Call the mock server with a client

This seems like it tests the auto-generated clients, but does not actually test our server implementation, correct?

It seems okay to have unit tests for the client, but I'm not sure I'd call these integration tests - we are not testing the integration of the client with the server, we're explicitly mocking the entire server.

Reply:
> This seems like it tests the auto-generated clients, but does not actually test our server implementation, correct?

This provides a mock for when Nexus makes HEAD requests as part of validating the URL sent in during DiskCreate.

Reply:
Ahhh makes sense. At first read, I thought this was mocking an API nexus was supporting, but it makes more sense as a mock *used* by nexus!